New article: The Christmas Time Machine

thibaultduponchelle opened the pull request:
Hello,

This is a new article about `Feature::Compat::Try`.

Preview:
![time-machine](https://user-images.githubusercontent.com/580360/186473087-602ea47f-1c34-4437-b5bd-583151ab2d8b.png)

simbabque commented on 2022/incoming/the-christmas-time-machine.pod:
```diff
@@ -0,0 +1,81 @@
+
+As usual they restarted doing line work, building toys, then sorting them per area. Everyone happily singing and working! When suddenly one elf put the first toy in a box marked "Salt Lake City" and got surprised it was not empty...
+
+And he got a terrible flashbak, they seemed to have forgotten to deliver 2021 toys in Salt Lake City!
```
- flashback, missing `c`
- to deliver _the_ 2021 toys _to_ Salt Lake City!

thibaultduponchelle replied:
Edited

simbabque commented on 2022/incoming/the-christmas-time-machine.pod:
```diff
@@ -0,0 +1,81 @@
+
+=head1 The Christmas time machine
+
+=head2 Terrible flashback at the line work
```
Can you please explain what the word flashback means? I understand this like in a movie, when you see a brief bit of what happened in the past. Is that what you mean?

thibaultduponchelle replied:
Exactly

openstrike replied:
"flashback" is fine for me, instead it is "at the line work" which reads oddly. Maybe it is an Americanism?

simbabque commented on 2022/incoming/the-christmas-time-machine.pod:
```diff
@@ -0,0 +1,81 @@
+
+=head2 Terrible flashback at the line work
+
+At North Pole, elves are just starting the Christmas 2022 preparations.
```
Pole, _the_ elves

thibaultduponchelle replied:
Edited

simbabque commented on 2022/incoming/the-christmas-time-machine.pod:
```diff
@@ -0,0 +1,81 @@
+
+=head2 Go ask Santa
+
+The elves were panicked, hence they reached quickly Santa for a solution. 
```
The elves _panicked_, _so_ they _quickly_ reached _out to_ Santa for a solution.

thibaultduponchelle replied:
Edited

simbabque commented on 2022/incoming/the-christmas-time-machine.pod:
```diff
@@ -0,0 +1,81 @@
+
+The elves were panicked, hence they reached quickly Santa for a solution. 
+
+He took some time to think about it then had an idea: Let's use the time machine to come back in 2021 and deliver the gifts!
```
- ... think about it_,_ then had
- to _go_ back _to_ 2021

thibaultduponchelle replied:
Edited

simbabque commented on 2022/incoming/the-christmas-time-machine.pod:
```diff
@@ -0,0 +1,81 @@
+
+(if you wonder: YES, time machines do exist in North Pole! And I bet that elves even already used it in the past for similar mistakes? :P)
+
+So a group of Elves decided to use the time machine, fixed the arrival date to 25 December 2021, pushed the "go" button and zooooooooooooop!
```
Is it elves or Elves? I feel capitalising is better, but you'll have to do it for all of them.

simbabque commented on 2022/incoming/the-christmas-time-machine.pod:
```diff
@@ -0,0 +1,81 @@
+
+But it failed with:
+
+	Feature "try" is not supported by Perl 5.30.0 at deliver.pl line 3.
```
Slight continuity problem... 5.30 was released in 2019. Would they not have 5.34 in 2021?

thibaultduponchelle replied:
I used the one that is shipped with Ubuntu in 2020 

Perl 5.34 has already the `try` feature...

simbabque reviewed (changes requested):
This is very nice! I did not realise where the story was going, like it a lot. I've pointed out a couple of typos and a bit of grammar. I'm not a native speaker, so maybe someone else can look it over as well.

thibaultduponchelle commented:
Thank you a lot @simbabque! I'm going to do the edits 👍🏼

oalders commented:
@thibaultduponchelle let me know when the edits are done and I'll look at this as well.

thibaultduponchelle commented:
@oalders edits done

openstrike commented on 2022/incoming/the-christmas-time-machine.pod:
```diff
@@ -0,0 +1,81 @@
+
+At North Pole, the elves are just starting the Christmas 2022 preparations.
+
+As usual they restarted doing line work, building toys, then sorting them per area. Everyone happily singing and working! When suddenly one elf put the first toy in a box marked "Salt Lake City" and got surprised it was not empty...
```
s/got/was/

thibaultduponchelle replied:
fixed

openstrike commented on 2022/incoming/the-christmas-time-machine.pod:
```diff
@@ -0,0 +1,81 @@
+
+As usual they restarted doing line work, building toys, then sorting them per area. Everyone happily singing and working! When suddenly one elf put the first toy in a box marked "Salt Lake City" and got surprised it was not empty...
+
+And he got a terrible flashback, they seemed to have forgotten to deliver the 2021 toys to Salt Lake City!
```
s/got/had/

openstrike commented on 2022/incoming/the-christmas-time-machine.pod:
```diff
@@ -0,0 +1,81 @@
+
+=head2 Go ask Santa
+
+The elves panicked, so they quickly reached out Santa for a solution. 
```
s/out Santa/out to Santa/

openstrike commented on 2022/incoming/the-christmas-time-machine.pod:
```diff
@@ -0,0 +1,81 @@
+
+The elves panicked, so they quickly reached out Santa for a solution. 
+
+He took some time to think about it, then had an idea: Let's use the time machine to go back in 2021 and deliver the gifts!
```
s/in 2021/to 2021/

thibaultduponchelle replied:
fixed

openstrike commented on 2022/incoming/the-christmas-time-machine.pod:
```diff
@@ -0,0 +1,81 @@
+
+=head2 Try to deliver back in the past
+
+When arrived in 2021, they immediately tried to run Perl software that they took with them.
```
Either "When they arrived" or "Upon arriving" would be a little better here.

thibaultduponchelle replied:
fixed

openstrike commented on 2022/incoming/the-christmas-time-machine.pod:
```diff
@@ -0,0 +1,81 @@
+
+When arrived in 2021, they immediately tried to run Perl software that they took with them.
+
+For the record, in 2021, North Pole was equiped with Ubuntu 20.04 and here is the Perl code they had:
```
"equipped" has 2 p's

thibaultduponchelle replied:
fixed

openstrike reviewed (changes requested):
There's nothing major wrong here AFAICS but a few minor fixes which would improve the overall reading of the article. HTH.

thibaultduponchelle commented:
Thank you a lot @openstrike I will do the edits 👍🏼

thibaultduponchelle commented:
Thank you a lot @openstrike for the corrections 👍🏼 And thank you again @simbabque for first round

@oalders I did the edits, ready for your review now 😃

oalders commented on 2022/incoming/the-christmas-time-machine.pod:
```diff
@@ -0,0 +1,81 @@
+Author: Thibault DUPONCHELLE <thibault.duponchelle@gmail.com>
+Title: The Christmas time machine
```
@thibaultduponchelle this is still outstanding ^^

oalders commented on 2022/incoming/the-christmas-time-machine.pod:
```diff
@@ -0,0 +1,81 @@
+
+=encoding utf8
+
+=head1 The Christmas time machine
```
```suggestion
=head1 The Christmas Time Machine
```

oalders commented on 2022/incoming/the-christmas-time-machine.pod:
```diff
@@ -0,0 +1,81 @@
+
+=head2 A terrible flashback
+
+At North Pole, the elves are just starting the Christmas 2022 preparations.
```
```suggestion
At the North Pole, the elves are just starting the Christmas 2022 preparations.
```

oalders commented on 2022/incoming/the-christmas-time-machine.pod:
```diff
@@ -0,0 +1,81 @@
+
+He took some time to think about it, then had an idea: Let's use the time machine to go back to 2021 and deliver the gifts!
+
+(if you wonder: YES, time machines do exist in North Pole! And I bet that elves even already used it in the past for similar mistakes? :P)
```
```suggestion
(If you're wondering: YES, time machines do exist in North Pole! And I bet that elves even already used it in the past for similar mistakes? :P)
```

oalders reviewed (changes requested):
Love it! Just a few small suggestions.

thibaultduponchelle commented:
Fixed all remarks, squashed.

Ready to merge! @oalders 

See the new preview:
![the-christmas-time-machine-preview](https://user-images.githubusercontent.com/580360/200508982-fcf04b1e-a7e3-407d-b424-648910c93206.png)

thibaultduponchelle commented:
Again, thank you a lot @simbabque and @openstrike 🙏🏼

oalders commented:
I'm going to close and re-open to see if that runs the test workflow.

oalders commented:
@openstrike and @simbabque, you'll need to approve the changes in order for us to merge this PR.

thibaultduponchelle commented:
@openstrike @simbabque can you please approve? 🙏🏼

oalders commented on 2022/incoming/the-christmas-time-machine.pod:
```diff
@@ -0,0 +1,81 @@
+Topic: Feature::Compat::Try
+
+=encoding utf8
+
+=head1 The Christmas Time Machine
```
```suggestion
```

oalders commented on 2022/incoming/the-christmas-time-machine.pod:
```diff
@@ -0,0 +1,81 @@
+
+=head1 The Christmas Time Machine
+
+=head2 A terrible flashback
```
```suggestion
=head2 A Terrible Flashback
```

oalders commented on 2022/incoming/the-christmas-time-machine.pod:
```diff
@@ -0,0 +1,81 @@
+
+And he had a terrible flashback, they seemed to have forgotten to deliver the 2021 toys to Salt Lake City!
+
+=head2 Go ask Santa
```
```suggestion
=head2 Go Ask Santa
```

oalders commented on 2022/incoming/the-christmas-time-machine.pod:
```diff
@@ -0,0 +1,81 @@
+
+So a group of elves decided to use the time machine, fixed the arrival date to 25 December 2021, pushed the "go" button and zooooooooooooop!
+
+=head2 Try to deliver back in the past
```
```suggestion
=head2 Try to Deliver Back in the Past
```
